File: app/src/main/java/com/rp/uihelpher/helpher/OnCall.java
package com.rp.uihelpher.helpher;

import android.app.Activity;
import android.content.Intent;
import android.net.Uri;

import com.rp.uihelpher.log.IsLog;

public class OnCall {

    private String TAG = OnCall.class.getSimpleName();

    public OnCall(Activity mActivity, String phoneNumber) {
        try {
            Intent dial = new Intent();
            dial.setAction("android.intent.action.DIAL");
            try {
                dial.setData(Uri.parse("tel:"+phoneNumber));
                mActivity.startActivity(dial);
            } catch (Exception e) {
                new IsLog("Calling", "" + e.getMessage());
            }
        }catch (NumberFormatException | NullPointerException e) {
            new IsLog(TAG,"NumberFormatException======"+e.getMessage());
        }catch (Exception e) {
            new IsLog(TAG,"Exception======"+e.getMessage());
        }
    }

}

File: app/src/main/java/com/rp/uihelpher/helpher/OnCall.kt
package com.rp.uihelpher.helpher

import android.app.Activity
import android.content.Intent
import android.net.Uri
import com.rp.uihelpher.log.IsLog
import androidx.core.net.toUri

class OnCall(mActivity: Activity?, phoneNumber: String?) {
    private val TAG: String = OnCall::class.java.simpleName

    init {
        try {
            val dial = Intent()
            dial.action = "android.intent.action.DIAL"
            try {
                dial.data = ("tel:$phoneNumber").toUri()
                mActivity?.startActivity(dial)
            } catch (e: Exception) {
                IsLog("Calling", "" + e.message)
            }
        } catch (e: NumberFormatException) {
            IsLog(TAG, "NumberFormatException======" + e.message)
        } catch (e: NullPointerException) {
            IsLog(TAG, "NumberFormatException======" + e.message)
        } catch (e: Exception) {
            IsLog(TAG, "Exception======" + e.message)
        }
    }
}
